src/praisonai/praisonai/scheduler/_delivery.py

```
@@ -93,6 +93,88 @@ def __init__(
+        v = self.validate()
+        if not v.ok:
+            logger.warning(
+                "Scheduler delivery target %r is not routable: %s %s",
+                self._deliver,
+                v.reason,
+                v.hint,
+            )
+        elif v.preview:
+            logger.info("Scheduled -> %s", v.preview)
+
+    def validate(self) -> "DeliveryValidation":
```

Ruff (0.16.1)

[error] Undefined name `DeliveryValidation`

(F821)